### ISSUE TITLE
await is ignored in `finally { await }` case

### ISSUE THREAD

**Issue by igisev** [rpetrich/babel-plugin-transform-async-to-promises#7]
Good day
`await` is not executed and does not wait for completion in the `finally` block

Test case:
```javascript
let messages = [];
let waitIndex = 0;

function wait() {
    let index = ++waitIndex;

    messages.push(`waitStart${index}`);

    return new Promise((resolve, reject) => setTimeout(resolve, 0))
        .then(() => {
            messages.push(`waitStop${index}`);
        });
}

async function test() {
    try {
        messages.push('tryStart');
        await wait();
        messages.push('tryStop');
    } catch (err) {
        messages.push('catchStart');
        await wait();
        messages.push('catchStop');
    } finally {
        messages.push('finallyStart');
        await wait();
        messages.push('finallyStop');
    }
}

messages.push('start');
test().then(() => {
    messages.push('stop');

    //  messages = ['start', 'tryStart', 'waitStart1', 'waitStop1', 'tryStop', 'finallyStart', 'finallyStop', 'stop']
    let expected = ['start', 'tryStart', 'waitStart1', 'waitStop1', 'tryStop', 'finallyStart', 'waitStart2', 'waitStop2', 'finallyStop', 'stop'];

    // false
    let isEqual = JSON.stringify(expected) === JSON.stringify(messages);

    console.log(isEqual, messages);
});
```

**Comment by igisev** [comment]
and case when `wait()` is always rejected

```javascript
let messages = [];
let waitIndex = 0;

function wait() {
    let index = ++waitIndex;

    messages.push(`waitStart${index}`);

    return new Promise((resolve, reject) => setTimeout(reject, 0))
        .catch(() => {
            messages.push(`waitStop${index}`);

            return Promise.reject(null);
        });
}

async function test() {
    try {
        messages.push('tryStart');
        await wait();
        messages.push('tryStop');
    } catch (err) {
        messages.push('catchStart');
        await wait();
        messages.push('catchStop');
    } finally {
        messages.push('finallyStart');
        await wait();
        messages.push('finallyStop');
    }
}

messages.push('start');
test().catch(() => {
    messages.push('stop');

    //  messages = ['start', 'tryStart', 'waitStart1', 'waitStop1', 'catchStart', 'waitStart2', 'waitStop2', 'finallyStart', 'finallyStop', 'stop']
    let expected = ['start', 'tryStart', 'waitStart1', 'waitStop1', 'catchStart', 'waitStart2', 'waitStop2', 'finallyStart', 'waitStart3', 'waitStop3', 'stop'];

    // false
    let isEqual = JSON.stringify(expected) === JSON.stringify(messages);

    console.log(isEqual, messages);
});
```

**Comment by rpetrich** [comment]
Fixed in v0.5.7
